### PR TITLE
Feature: add a parameter to /profile and /profiles to get its metadata

### PR DESCRIPTION
**Pull request by lsenta**

### CODE REVIEW

File: src/pinning.js [hunk 1]
```diff
@@ -56,8 +56,13 @@ class Pinning {
           })
         const profileFromPubStore = address => {
           const profile = this.openDBs[address].db.all()
-          let parsedProfile = {}
-          Object.keys(profile).map(key => { parsedProfile[key] = profile[key].value })
+          const parsedProfile = {}
+
+          Object.entries(profile)
```
Comment (oed) [inline src/pinning.js]: Is this really needed? Isn't the format already `{ value, timeStamp }`?

File: src/cacheService.js [hunk 2]
```diff
@@ -138,8 +138,36 @@ class CacheService {
     }
   }
 
+  /**
+   * Process back the profile into a form that depends whether the user queried
+   * for metadata or not.
+   *
+   * Returns either the profile
+   * - WITH metadata: `{name: {timestamp: 123123123123, value: "Dvorak"}, ...}`
+   * - WITHOUT metadata: `{name: "Dvorak", ...}`
```
[the remark this answers is not in the document]
Comment (lsenta) [reply src/cacheService.js]: Yea, I started with `__metadata_` (first commit) but I realized, the API will be nicer to use and simpler to extend with something like:

```
GET /profile?addr=0x...&metadata=value,timestamp,origin,etc

=>

{
  name: { value: ..., timestamp: ..., origin: ...., etc: ....},
  age: { value: ..., timestamp: ..., origin: ...., etc: ....},
}
```

File: src/pinning.js (hunk 1 above)
Comment (lsenta) [reply src/pinning.js]: There's a unusual capitalization on timeStamp, I return `{ value, timestamp }`

Comment (oed) [reply src/pinning.js]: Oh, I see you did the same thing in the js lib as well. Is it really worth the extra complexity though?

**Comment by zachferland** [comment]
it will still be timeStamp in the lib though when not using the api right?

**Comment by lsenta** [comment]
@zachferland It's consistent, `timestamp` too: https://github.com/3box/3box-js/blob/develop/src/keyValueStore.js#L31

File: src/pinning.js (hunk 1 above)
Comment (lsenta) [reply src/pinning.js]: (discussed on discord: same time complexity as before)

**Comment by zachferland** [comment]
was that case before or as of recently ? if new, that's a breaking change in the lib then?

**Comment by lsenta** [comment]
@zachferland No breaking changes, It's a new feature that lets user access metadata.

Here's the full EPIC:
https://app.zenhub.com/workspaces/3box-5b37f0a38968796fbdda5e06/issues/3box/3box/524

**Comment by zachferland** [comment]
right, I see, yeah our dapp used the underlying log where it is timeStamp, but see now this was for new  func 👍

**Pull request merged**